app/services/order_purchaser.rb

```diff
@@ -79,12 +79,19 @@ def quantities
     order.order_details.order("order_details.id").pluck(:quantity)
   end
 
-  def send_facility_notification?
-    order_notification_recipient.present? && !acting_as?
-  end
+  def send_purchase_notifications
+    if !acting_as? || params[:send_notification] == "1"
```

Comment:
Hmm, I'm definitely in the school of skinny methods and reusable pieces, even though at the moment we might not actually be reusing the pieces elsewhere. I won't insist on it though.

My biggest issue with this method in its current form is that it's hard for me to tell at a glance what each piece is supposed to be doing. Whatever you want to do to clarify the function of each chunk would be great, whether that's using local variables or just adding some code comments.

Re sending product-level notifications--I'm not sure. I would default towards sending the notification even if acting-as, because I would rather they tell us they don't need the notification than have them be upset because they were relying on the notification and didn't get it.  Also, f there are multiple staff members for a facility--I imagine the person who places the order on behalf of might not be the same person who would be expecting the product specific notification.